### PR TITLE
Gayatri

### PR DESCRIPTION
**Pull request by GAYATRIBHAGWAT11**
data table is completed

### CODE REVIEW

File: src/Component/DataTable/DataTable.jsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState } from "react";
+import '/Users/admin/CovidAppProject/COVID-APP-GROUP-R1/src/Component/DataTable/DataTable.css'
+import Posts from "../DataTable/Posts";
```
Comment (Souvikdeb2612) [inline src/Component/DataTable/DataTable.jsx:6]: First external imports occur that is imports from installed libraries, then 1 line space, then all internal imports that is files/components/functions that you have created, then 1 line space and then finally all css imports.

File: src/Component/DataTable/DataTable.jsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            return null;
+        }
+    });
+
+
+    useEffect(() => {
+        setLoading(true)
+        axios
+            .get(
+                'https://vaccovid-coronavirus-vaccine-and-treatment-tracker.p.rapidapi.com/api/npm-covid-data/',
```
Comment (Souvikdeb2612) [inline src/Component/DataTable/DataTable.jsx]: store url externally in a state/variable and use wherever required

File: src/Component/DataTable/DataTable.jsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                    },
+                }
+            )
+            .then(res => {
+                console.log(res.data);
+                setDataTab([...res.data]);
+                setLoading(false)
+            })
+            .catch(function (err) {
+                console.error(err);
```
Comment (Souvikdeb2612) [inline src/Component/DataTable/DataTable.jsx]: remove console statements before raising pr

File: src/Component/DataTable/DataTable.jsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                'https://vaccovid-coronavirus-vaccine-and-treatment-tracker.p.rapidapi.com/api/npm-covid-data/',
+                {
+                    headers: {
+                        'X-RapidAPI-Key': '69b56f0544mshe1d9f75b422a4e7p142f81jsn40c4bc0c5681',
+                        'X-RapidAPI-Host': 'vaccovid-coronavirus-vaccine-and-treatment-tracker.p.rapidapi.com'
+                    },
+                }
+            )
+            .then(res => {
+                console.log(res.data);
```
Comment (Souvikdeb2612) [inline src/Component/DataTable/DataTable.jsx]: remove console statements before raising pr

File: src/Component/DataTable/DataTable.jsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+                    <ul className="Pagenumbers">
+                        <li>
+                            <button className="btn" onClick={handlePrevBtn} disabled={currentPage == pages[0] ? true : false}>Prev</button>
+                        </li>
+                        {pageDecrementBtn}
+                        {renderPageNumbers}
+                        {pageIncrementBtn}
+
+                        <li>
+                            <button className="btn" onClick={handleNextBtn} disabled={currentPage == pages[pages.length - 1] ? true : false}>Next</button>
```
Comment (Souvikdeb2612) [inline src/Component/DataTable/DataTable.jsx]: create a function for the disabled feature and call the function here

File: src/Component/DataTable/DataTable.jsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+            <div className="container">
+                <h1>All Country Data</h1>
+                <Posts posts={currentPosts} loading={loading} />
+
+                {/* <Pagination postsPerPage={postsPerPage} totalPosts={dataTab.length} paginate={Paginate} /> */}
+                <div className="dataCont">
+                    <ul className="Pagenumbers">
+                        <li>
+                            <button className="btn" onClick={handlePrevBtn} disabled={currentPage == pages[0] ? true : false}>Prev</button>
```
Comment (Souvikdeb2612) [inline src/Component/DataTable/DataTable.jsx]: create a function for the disabled feature and call the function here

File: src/Component/DataTable/Posts.jsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
+
+import React from "react";
+import '/Users/admin/CovidAppProject/COVID-APP-GROUP-R1/src/Component/DataTable/Posts.css'
```
Comment (Souvikdeb2612) [inline src/Component/DataTable/Posts.jsx]: add 1 line space before css imports

File: src/Component/DataTable/Posts.jsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+                            <th>cases(per million)</th>
+                            <th>Total cases</th>
+                            <th>Total Deaths</th>
+                            <th>Total Recovered</th>
+                            <th>Total Tests</th>
+
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {posts.map(post => {
```
Comment (Souvikdeb2612) [inline src/Component/DataTable/Posts.jsx]: destructure post above

File: src/Component/DataTable/Posts.jsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+                            return <tr>
+                                <td>{post.Country}</td>
+                                <td>{post.Population}</td>
+                                <td>{post.ActiveCases}</td>
+                                <td>{post.TotCases_1M_Pop}</td>
+                                <td>{post.TotalCases}</td>
+                                <td>{post.TotalDeaths}</td>
+                                <td>{post.TotalRecovered}</td>
+                                <td>{post.TotalRecovered}</td>
```
Comment (Souvikdeb2612) [inline src/Component/DataTable/Posts.jsx]: total recovered is coming twice

File: src/Component/DataTable/Posts.jsx [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        return <h2>Loading.....</h2>
+    }
+    // console.log(posts);
+    return (
+        <>
+            <div className="Data-cont">
+                <table key={posts.id} id="cd">
+                    <thead>
+                        <tr>
+                            <th>Country</th>
```
Comment (Souvikdeb2612) [inline src/Component/DataTable/Posts.jsx:15]: This is constant data so export from constant folder

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

**Comment by GAYATRIBHAGWAT11** [comment]
changes resolved

**Review by GAYATRIBHAGWAT11 — COMMENTED** [review]
resolved changes

**Comment by GAYATRIBHAGWAT11** [comment]
registration form completed

**Pull request merged**